src/dotty/tools/dotc/transform/InterceptedMethods.scala

```diff
@@ -103,7 +103,13 @@ class InterceptedMethods extends MiniPhaseTransform {
       case Any_## =>
           poundPoundValue(qual)
       case Any_!= =>
-          qual.select(defn.Any_==).appliedToArgs(tree.args).select(defn.Boolean_!)
+        val lhs = qual.tpe
+        val rhs = tree.args.head.tpe
+        if (!(lhs <:< rhs || rhs <:< lhs)) {
+          ctx.warning(ex"comparing values of types ${lhs.widenDealias} and ${rhs.widenDealias} using `!=' will always yield true",
+              tree.pos)
+        }
+        qual.select(defn.Any_==).appliedToArgs(tree.args).select(defn.Boolean_!)
```

Comment:
Yes, the tests is incomplete, in scalac there is a different waring for those cases:
```
scala> x != y
<console>:18: warning: A and B are unrelated: they will most likely always compare unequal
       x != y
```

Now I'm convinced that these should not be a separate mechanism from multiversal equality.

There is also the question of the `NullType <:< X` tests failing for `X`s that are term refs to some subtype of `AnyRef`. In all the currently tests those term refs are widen first and therefore we never have false negatives. Is this the correct way to handle those cases or the change to `isNullabe` should be done?